microsoft/surface/old/default.nix

```diff
@@ -0,0 +1,22 @@
+{ lib, pkgs, ... }:
+
+let
+  inherit (lib) mkDefault;
+
+in {
+  imports = [
+    ../common
+    ../surface-go/firmware/ath10k
+  ];
+
+  boot.extraModprobeConfig = mkDefault ''
+    options i915 enable_fbc=1 enable_rc6=1 modeset=1
+    options snd_hda_intel power_save=1
+    options snd_ac97_codec power_save=1
+    options iwlwifi power_save=Y
+    options iwldvm force_cam=N
+  '';
```

Comment:
Given this module still module still has some option is this maybe some generic intel model, that you trying to target here?

Reply:
I think as of now it's a bit unclear to where people have to migrate after switching to the old module?

Reply:
Are you trying to encourage people to add their own more specific models instead of using the `old` module?

Reply:
Ah, damn:  I had intended to add doc-comments to this file, to explain its intention as well as the `old/README.md`.
I will add that.

The intention is to support the "old behaviour" that most people have probably been relying on, and provide an additional deprecation warning asking people to change.
The old behaviour included everything Surface related, and expected various kernel modules and other Surface support tools to fail gracefully depending on the hardware they were running on.

I feel like the change is big enough that having some sort-of deprecation process is appropriate.

Reply:
> The intention is to support the "old behaviour" that most people have probably been relying on, and provide an additional deprecation warning asking people to change. The old behaviour included everything Surface related, and expected various kernel modules and other Surface support tools to fail gracefully depending on the hardware they were running on.
> [...]
> I feel like the change is big enough that having some sort-of deprecation process is appropriate.

Now that I've had a brain-break, I'm thinking a better option would be to drop the `deprecated` module, and instead make the original `microsoft/surface/default.nix` fail with an explanation with how to update.

Is there a way to support this, and still make sure the CI tests succeed?